### PR TITLE
Testool take II

### LINKED ISSUE #783: Define subset of eth test-tool tests for sanity checks during CI.

### PR DESCRIPTION
**Pull request by adria0**
In this version of testool

- Better config file
- `light` suite to execute one test per opcode (to integrate into CI)
- Able to specify full tests from command line using `--oneliner`
- Able to run the supercircuit

### CODE REVIEW

File: testool/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+- `account` have the form `address`;`code`;`balance`;`slot`1:`value`1;..`slot`n..`value`n
+  - `address` can be shortened, eg.`10` and `0x10` is expanded to `0x....10`
+  - `code` is optional, and can be specified in hex or in asm (`PUSH1(1),PUSH1(1),MLOAD`)
+  - `balance` and `value` areoptional
+  - `slot` and `value` can be specified multiple times 
+
+example
+
+- `call 12;60016002`: call contract `0x...12` that contains the code PUSH1(1) PUSH1(2)
+- `call;;2000 12;PUSH1(0),SLOAD,CALLVALUE,EQ,PUSH1(1),SSTORE;;00:2000`: callthe contract and send 2000 as value, and compare with the stored value (2000) in the slot 0, write into slot 1
```
Comment (ChihChengLiang) [inline testool/README.md:31]: It took me a while to find the right way to run the correct command. What do you think we write the verbose form here?
```suggestion
- `../target/release/testool --oneliner "call 12;60016002"`: call contract `0x...12` that contains the code PUSH1(1) PUSH1(2)
- `../target/release/testool --oneliner "call;;2000 12;PUSH1(0),SLOAD,CALLVALUE,EQ,PUSH1(1),SSTORE;;00:2000"`: call the contract and send 2000 as value, and compare with the stored value (2000) in the slot 0, write into slot 1 
```

File: testool/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+This tool at this moment has 2 main functionalities: run raw bytecode and run ethereum tests.
+
+## Run oneliner spec
+
+The oneliner spec is invoked by using `--oneliner` parameter
+
+The spec has the following form
+
+`txparams` `account`1 `account`2 ... `account`n
```
Comment (ChihChengLiang) [inline testool/README.md:15]:
```suggestion
`txparams` `account 1` `account 2` ... `account n`
```

File: testool/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+example
+
+- `call 12;60016002`: call contract `0x...12` that contains the code PUSH1(1) PUSH1(2)
+- `call;;2000 12;PUSH1(0),SLOAD,CALLVALUE,EQ,PUSH1(1),SSTORE;;00:2000`: callthe contract and send 2000 as value, and compare with the stored value (2000) in the slot 0, write into slot 1 
+
+## Run the ethereum tests
+
+The "official EVM" ethereum tests are cloned as a gitmodule in `testool/tests`.
+We are using the tests located in `testtool/tests/src/GeneralStateTestsFiller`, but other locations can be specified, also.
```
Comment (ChihChengLiang) [inline testool/README.md:36]: Can you give an example of how to run this?

File: testool/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+There are two ways to execute the tests, in "debug mode
```
Comment (ChihChengLiang) [inline testool/README.md:38]: Is this sentence unfinished?

**Review by ChihChengLiang — COMMENTED** [review]
Attempted to run the command from the instruction. But only the one-liner works for me.
I got "Suite not found" for other commands. I have the submodule updated and the tests submodule exists.

Would be great to have a `ls` command to list tests available to run.

**Review by ed255 — COMMENTED** [review]
I've done a fast review and everything looks good to me!  I've added some change suggestions for the README related to typos.

**Review by ChihChengLiang — APPROVED** [review]
LGTM after issues are addressed.

**Pull request merged**